lib/syskit/test/network_manipulation.rb

```diff
@@ -227,31 +238,32 @@ def syskit_deploy_normalized_placeholder_tasks(
                     .to { emit(*not_running.map(&:start_event)) }
 
                 resolve_options = Hash[on_error: :commit].merge(resolve_options)
-                begin
+                resolution_errors = execute do
                     syskit_engine_resolve_handle_plan_export do
                         syskit_engine ||= Syskit::NetworkGeneration::Engine.new(plan)
                         syskit_engine.resolve(
                             default_deployment_group: default_deployment_group,
                             **resolve_options
                         )
                     end
-                rescue StandardError => e
-                    expect_execution do
-                        requirement_tasks.each { |t| t.failed_event.emit(e) }
-                    end.to do
+                end
+                if resolution_errors.empty?
+                    execute do
+                        placeholder_tasks.each do |task|
+                            plan.remove_task(task)
+                        end
                         requirement_tasks.each do |t|
-                            have_error_matching(Roby::PlanningFailedError
-                                                .match.with_origin(t.planned_task))
+                            t.success_event.emit unless t.finished?
                         end
                     end
-                    raise
-                end
-
-                execute do
-                    placeholder_tasks.each do |task|
-                        plan.remove_task(task)
+                else
+                    resolution_errors.each do |error|
+                        t = error.planned_task
+                        expect_execution do
+                            t.failed_event.emit(error.original_exception)
```

Comment:
You've got a problem here if the task was from a previous deploy (can't emit failed_event)